src/Transition/CreateNewIDFUsingRulesV23_2_0.f90

```diff
@@ -674,9 +674,35 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                 OutArgs(20:CurArgs+1)=InArgs(19:CurArgs)
                 CurArgs = CurArgs + 1
 
-              ! If your original object starts with D, insert the rules here
+              ! If your original object starts with D, insert the rules here              
+              CASE('DISTRICTHEATING')
+                nodiff=.false.
+                ObjectName='DistrictHeating:Water'
+                CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                OutArgs(1:CurArgs)=InArgs(1:CurArgs)
 
               ! If your original object starts with E, insert the rules here
+              CASE('ENERGYMANAGEMENTSYSTEM:METEREDOUTPUTVARIABLE')
+                 CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                 nodiff=.false.
+                 OutArgs(1:CurArgs)=InArgs(1:CurArgs)
+                 IF (OutArgs(5) == "DISTRICTHEATING" .OR. OutArgs(5) == "DistrictHeating" .OR. OutArgs(5) == "districtheating") THEN
+                     OutArgs(5) = "DistrictHeatingWater"
+                 END IF
+                 IF (OutArgs(5) == "STEAM" .OR. OutArgs(5) == "Steam" .OR. OutArgs(5) == "steam") THEN
+                     OutArgs(5) = "DistrictHeatingSteam"
+                 END IF
```

Comment:
This object has min-fields 7 so this field should always be there.  However, the upper/lower case needs work here.  I can't recall the Fortran string comparison with trailing spaces.  OutArgs(5) is actually like 100 characters long filled with blanks.  So at a minimum I think we need to TRIM the OutArgs(5) to be sure we are comparing just the non-blank characters.  But more than that, what if the user enters DISTRICTHeating?  That's not a case here.  We should do something different -- I think Transition has SameString which will handle all these things nicely.  

```
IF (SameString(OutArgs(5), "DistrictHeating") THEN
```

should work I think.